public/js/angular/directives/wysiwyg.js

```diff
@@ -0,0 +1,405 @@
+          scope.restoreSelection();
+
+          var link = '<a href="' + scope.layoutLink.url + '" ' + (scope.layoutLink.newTab ? 'target="_blank"' : '') + '>' + scope.layoutLink.text + '</a>';
+          scope.formatAction('inserthtml', link);
+        };
+
+        scope.insertReadMore = function() {
+          scope.restoreSelection();
+          scope.formatAction('inserthtml', '<hr class="read_more_break"></hr>');
+        };
+
+        scope.mediaAvailable = function() {
+          if(scope.media && scope.media instanceof Array) {
+            return true;
+          }
+
+          return false;
+        };
+
+        scope.showInsertMediaModal = function() {
+          angular.element(element).find('[contenteditable]').focus();
+          scope.saveSelection();
+          angular.element(element).find('[insert-media-modal]').modal('show');
+        };
+
+        scope.associateMedia = function() {
+          angular.element(element).find('[insert-media-modal]').modal('hide');
+          angular.element('.nav-tabs a[href="#media"]').tab('show');
+        };
+
+        scope.insertMedia = function() {
+          var mediaFormat = scope.getMediaFormat();
+          scope.formatAction('inserthtml', '<div>^media_display_' + scope.wysiwyg.selectedMediaItem._id + mediaFormat + '^</div>');
+
+          angular.element(element).find('[insert-media-modal]').modal('hide');
+          scope.restoreSelection();
+          scope.wysiwyg.layout = angular.element(element).find('[contenteditable]').html();
+        };
+
+        scope.getMediaFormat = function() {
+          var mediaFormat = '/position:' + scope.wysiwyg.mediaPosition;
+
+          if(scope.wysiwyg.mediaMaxHeight) {
+            mediaFormat = mediaFormat.concat(',maxheight:' + scope.wysiwyg.mediaMaxHeight + scope.wysiwyg.mediaMaxHeightUnit);
+          }
+
+          return mediaFormat;
+        }
+
+        scope.saveSelection = function() {
+          if(!angular.element(element).find('[contenteditable]').is(':focus')) {
+            return;
+          }
+
+          if(scope.editableSelection) {
+            rangy.removeMarkers(scope.editableSelection);
+          }
+          scope.editableSelection = rangy.saveSelection();
+        };
+
+        scope.restoreSelection = function() {
+          if(scope.editableSelection) {
+            rangy.restoreSelection(scope.editableSelection, true);
+            scope.editableSelection = null;
+          }
+        };
+
+        scope.getSelection = function() {
+          if(scope.editableSelection) {
+            return rangy.getSelection().toString();
+          }
+
+          return '';
+        }
+
+        scope.loadMediaPreviews = function() {
+          if(scope.wysiwyg.currentView !== 'editable') {
+            return;
+          }
+
+          var index = scope.wysiwyg.layout.indexOf('^media_display_');
+          if(index === -1) {
+            return;
+          }
+
+          var startIndex = index + 15;
```

Comment:
make the prefix ```^media_display_``` a constant at the top of the file then reference by ".length" rather than magic number.